cjworkbench/i18n/trans.py

```diff
@@ -174,25 +199,25 @@ def _replace_tags(self, target_message, tags):
 
     def _format_message(self, message, parameters={}, do_escape=True):
         """Substitute parameters into ICU-style message.
-        At this point, ICU is not actually supported (i.e. you can have no plurals, secects, etc).
-        Only variable substitution is supported.
+        You can have variable substitution, plurals, selects and nested messages.
         
         The parameters must be a dict
         """
         if not message:
             return message
+        message_format = MessageFormat(
+            UnicodeString(message), Locale.createFromName(self.locale)
+        )
         if isinstance(parameters, dict):
             try:
-                return formatter.format(
-                    message,
-                    **{
-                        key: (
-                            escape(parameters[key])
-                            if do_escape and isinstance(parameters[key], str)
-                            else parameters[key]
+                return message_format.format(
+                    [UnicodeString(x) for x in list(parameters.keys())],
```

Comment:
I would think this could simply be `list(parameters.keys())` -- no list comprehension.